include/view_as_pointer.h

```diff
@@ -35,8 +37,8 @@
 //   dereferenced, it should not be an issue.
 //
 
-template<class T>
-void *view_as_pointer(const T &val)
+template<class T, class = typename std::enable_if<std::is_integral<T>::value>::type>
```

Comment:
Nice catch!!!

That's pretty cool. Going to have to remember that. Here is a reference if anyone wants to know how this black magic works. 

https://en.wikibooks.org/wiki/More_C%2B%2B_Idioms/SFINAE

Now all that being said, I really wish they would have just added this to the language so the syntax was easier to understand. These sort of tricks work, but they are confusing.